flink-table/flink-table-runtime/src/test/java/org/apache/flink/table/runtime/functions/table/fullcache/inputformat/InputFormatCacheLoaderTest.java

```diff
@@ -149,19 +155,33 @@ void testCloseAndInterruptDuringReload() throws Exception {
         Future<?> future = executorService.submit(cacheLoader);
         executorService.shutdownNow(); // internally interrupts a thread
         assertThatNoException().isThrownBy(future::get); // wait for the end
-        // check that we didn't process all elements, but reacted on interruption
-        assertThat(sleepCounter).hasValueLessThan(totalSleepCount);
         assertThat(metricGroup.numLoadFailuresCounter.getCount()).isEqualTo(0);
+    }
 
-        sleepCounter.set(0);
+    @Test
+    void testCloseDuringReload() throws Exception {
+        AtomicInteger recordsCounter = new AtomicInteger(0);
+        int totalRecords = TestCacheLoader.DATA.size() + 1; // 1 key with 2 records
```

Comment:
```suggestion
        int totalRecords = TestCacheLoader.DATA.values().stream().mapToInt(Collection::size).sum();
```
nit: you could use this call. That way you could remove the comment which easily becomes wrong if someone changes `TestCacheLoader.DATA`. WDYT?

Reply:
Agree, fixed